indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -310,6 +351,146 @@ public TaskLock lock(final Task task, final Interval interval) throws Interrupte
   }
 
   /**
+   * Check priorities of existing tasks, if all has lesser priorities and no lock is upgraded then return true
+   * If this task can acquire lock and some lower priority TaskLocks need to be revoked then populate tasksToPreempt Map
+   * with the entry - taskId and TaskLock
+   * @return -
+   *  (true, Map<taskId, taskLock>) if there are no active tasks having lockPriority >= {@param lockPriority},
+   *    map contains taskIds and associated lower priority taskLocks that needs to be revoked
+   *  (false, Map<taskId, taskLock>) if there are some active tasks having lockPriority >= {@param lockPriority},
```

Comment:
Here we may want to revoke locks of the lower priority tasks or there are no tasks holding conflicting task lock or there are some conflicting tasklock that have higher priority so simply returning the list of conflicting tasks will not work. Am I missing something ?